### ISSUE TITLE
Second Shield fields

### ISSUE THREAD

**Issue by Flametix** [SteveCookTU/sv_raid_lookup#4]
Based on past behavior from 7 star raids, the fields currently named "Second Shield Trigger HP/Time/Damage Rate" seem to determine when the boss starts being able to move twice in a turn and the chance of the second action happening.

**Comment by SteveCookTU** [comment]
Hey there! Thanks for pointing this out. Do you have an example of this? I wanted to test it out but the current raids currently have dummy data for those fields.

**Comment by Flametix** [comment]
I remember that at least Decidueye, Delphox, and Rillaboom had this come up. Here's an image I saved of Rillaboom's set showing it would activate under 55% HP.
![1690508141278](https://github.com/SteveCookTU/sv_raid_lookup/assets/44883711/b0a83566-c09d-45bb-9685-22e5733c4782)

Here's a video of an offline solo, on the turn at 6:53 Rillaboom gets to use Acrobatics and Body Slam in the same turn.

https://youtu.be/q_LEAYDLAMA?t=399

**Comment by SteveCookTU** [comment]
Unless this was changed, to my knowledge this field was originally for triggering a second shield. It's been awhile since I've interacted with 7 star raids but I remember the first one having major bugs because of the second shield. The names for the fields are how they have them labeled in their schema. I'm going to keep them as-is and keep this issue open as more information comes to light.

**Comment by Flametix** [comment]
The latest Hisuian Decidueye raids have a 100% chance of double acting after a certain HP threshold and it's pretty noticeable there. I think what you're thinking of with the Charizard raid was the Play Rough glitch making Charizard appear to regenerate its shield after it was broken, because Play Rough was visually doing more damage than it actually did and was only corrected after the turn.

I also noticed in sv_raid_reader that you have the field referenced as "doubleActionTrigerHP" there too which I assume is based on the internal names for the fields.
https://github.com/SteveCookTU/sv_raid_reader/blob/08cb5a24a2ad6be9760e77b182ac099eb08bcfb2/src/raid_encounter.rs#L491

On the topic of field names, I think it would be more clear to say that "Shield Gem Damage Rate" is the damage rate for hitting a shield with the same type move and tera type, while "Shield Change Gem Damage Rate" is the rate for hitting a shield while terastallized with a move not of the same type as your tera. 
https://twitter.com/Sibuna_Switch/status/1610483831769018368?lang=en

**Comment by SteveCookTU** [comment]
Good catch, the double action is the internal name which makes sense then for it to be what you mean. 

Also you are correct, the last two fields you mention are damage based on type when tera'd
